CMakeLists.txt

```diff
@@ -1488,6 +1488,8 @@ set(CPACK_RPM_FILE_NAME RPM-DEFAULT)
 
 set(CPACK_ARCHIVE_COMPONENT_INSTALL ON)
 
+set(CPACK_RPM_SPEC_MORE_DEFINE "%global _python_bytecompile_errors_terminate_build 0")
```

Comment:
Are there any downsides to this?

Reply:
Ehhh, kinda, but nothing we can really do anything about while still using CPack AFAIK.

My understanding is that stable contains a mix of python 2 and python 3 code, and some .py files that are compatible with both; and that we are still primarily using python 2 in stable.

The documentation for how rpmbuild's automatic byte compilation works is pretty sparse, but based on what I could find, and what I could suss out from looking at source code, I've made the following observations:
- Without doing the byte compilation ourselves, we can only byte-compile for one version of Python.
- Automatic btye compilation is *supposed* to only happen for .py files in standard locations. Our stuff in `/var/lib/irods` is getting picked up for byte compilation, which means one of the following:
    - The documentation is lying about this.
    - The byte compiler script has a bug.
- Python will sometimes do its own byte compilation if pre-compiled .pyc/.pyo files are not found at runtime.
    - The package manager cannot track these files, which could cause problems when upgrading or uninstalling.

Since we are still primarily using Python 2 in stable, I felt the least-bad solution was to *continue using Python 2 for byte compilation* and allow the packaging to proceed if a .py file fails to compile.
This means that if Python *3* ever runs a .py file we package, it may generate .pyc/.pyo files that the package manager does not track. *However,* I feel that this will come up infrequently enough that we can deal with it on a case-by-case basis, should problems arise.